src/utils/bitcoin.ts

```diff
@@ -83,7 +83,7 @@ export function decodeBtcAddress(
 }
 
 export async function getTxProof(
```

Comment:
I think this function can be simplified:
1. Require `btcTxId` as a parameter then you don't need the extra if condition and throw an error. It's just required on function call. If you already have both the merkleProof and rawTx, you'd anyway not call that function.
2. Default to some electrsAPI OR make it a required parameter. Having an optional parameter that just throws if not supplied is bad UX for the API.

Reply:
I refactored this to take the `TxFetchingDetails` type, let me know what you think